include/heuristic/HeuristicMapper.hpp

```diff
@@ -151,6 +193,99 @@ class HeuristicMapper : public Mapper {
       }
 
       swaps.back().emplace_back(source, target, middleAnc, qc::Teleportation);
+
+      if (considerFidelity) {
+        throw QMAPException(
+            "Teleportation currently not supported for noise-aware mapping");
+        // TODO: check this cost implementation and add teleportation support in
+        // noise-aware heuristic accounting for fidelity difference of single
+        // qubit gates (two qubit gates are handled in the heuristic)
+        /*costFixed += (-(singleQubitGateMultiplicity.at(q2) -
+        singleQubitGateMultiplicity.at(q1)) *
+                         log2(singleQubitFidelities.at(swap.first))
+                      -(singleQubitGateMultiplicity.at(q1) -
+        singleQubitGateMultiplicity.at(q2)) *
+                        log2(singleQubitFidelities.at(swap.second))
+                      );
+        // adding cost oft the swap gate
+        costFixed += (-    log2(twoQubitFidelities.at(source).at(middleAnc))
+                      -    log2(twoQubitFidelities.at(middleAnc).at(target))
+                      -    log2(singleQubitFidelities.at(source))
+                      -    log2(singleQubitFidelities.at(middleAnc))
+                      -2 * log2(singleQubitFidelities.at(target))
+                      -    log2(1.0 -
+        arch.getProperties().readoutErrorRate.get(source))
+                      -    log2(1.0 -
+        arch.getProperties().readoutErrorRate.get(middleAnc)));*/
+      } else {
+        costFixed += COST_TELEPORTATION;
+      }
+    }
+
+    /**
+     * @brief recalculates the fixed cost of the node from current mapping and
+     * swaps
+     *
+     * @param arch the architecture for calculating distances between physical
+     * qubits and supplying qubit information such as fidelity
+     * @param singleQubitGateMultiplicity vector containing the number of gates
+     * acting on each logical qubit
+     * @param considerFidelity controls if qubit fidelity should be taken into
+     * account
+     */
+    void recalculateFixedCost(
```

Comment:
I'd suggest moving the definition of the function to the source file. In order to keep the header file "clean". Probably applies to some of the other functions as well.